spec/unit/rom/gateway_spec.rb

```diff
@@ -87,4 +87,38 @@ class Gateway
       expect(gateway.disconnect).to be(nil)
     end
   end
+
+  describe '.adapter' do
+    let(:gateway_class) { Class.new(ROM::Gateway) }
+
+    it 'gets/sets the adapter' do
+      expect { gateway_class.adapter :custom }
+        .to change { gateway_class.adapter }
+        .from(nil)
+        .to(:custom)
+    end
+  end
+
+  describe "#adapter" do
+    subject { ROM::CustomAdapter.new.adapter }
```

Comment:
Please don't use anonymous subjects like that, `subject` is meant to be used to set "object under test". Here you set it to a method call. This should be:

``` ruby
subject(:gateway) { ROM::CustomAdapter.new }

# and then
it 'requires the adapter to be defined' do
  expect { gateway.adapter }.to raise_error(ROM::MissingAdapterIdentifierError, /ROM::CustomAdapter/)
end
```

I know we've been writing a ton of specs with implicit subjects and inlined examples but it's history, those specs were difficult to maintain and hard to understand and follow for people.